debugging pings (#858)
sourcegraph/about: handbook/bizops/index.md

We have [written policies about how we handle customer information](customer_data_policy.md). 

## Debugging pings

> Use this section to debug ping abnormalities (from top to bottom).

Reviewer: I'm not sure if the order someone should follow would always be "top to bottom". My hunch is that different signals in ping failures could lead us to want to test from one end first.

In general, it'd be nice to refer back to the "life of a ping" doc (or perhaps we move it to the handbook, or we put this section in the docs? I'm not totally convinced either way...). Then we can use this list as a guide for "how to debug each step" rather than a prescriptive "here's the order to debug in". 

What do you think?

Reviewer: Also, this probably makes sense to move to a new file — this page is getting pretty dense :). Happy to help with this if that'd be helpful!

Author: @dadlerj agree with those suggestions. I've been thinking about changing the architecture of the page; I'll address that in a separate PR 👍

1. BigQuery: Query the [update_checks error records](https://console.cloud.google.com/bigquery?sq=839055276916:62219ea9d95d4a49880e661318f419ba) and/or [check the latest pings received](https://console.cloud.google.com/bigquery?sq=839055276916:3c6a5282e66a4f0fac1b958305d7b197) based on installer email admin. 
1. Dataflow: Review [Dataflow](https://console.cloud.google.com/dataflow/jobs/us-central1/2020-02-05_10_31_47-13247700157778222556?project=telligentsourcegraph&organizationId=1006954638239): WriteSuccessfulRecords should be full of throughputs and the Failed/Error jobs should be empty of throughputs. 
1. Stackdriver (log viewer): [Check the frontend logs](https://console.cloud.google.com/logs/viewer?project=sourcegraph-dev&minLogLevel=0&expandAll=false&customFacets=&limitCustomFacetWidth=true&interval=PT1H&resource=k8s_container%2Fcluster_name%2Fdot-com%2Fnamespace_name%2Fprod%2Fcontainer_name%2Ffrontend), which contain all pings that come through Sourcegraph.com. Use the following the advanced filters to find the pings you're interested in.

> 

Reviewer: Let's also add a bullet for "test on a Sourcegraph dev instance" to make sure the pings are being sent properly

We have [written policies about how we handle customer information](customer_data_policy.md). 

## Debugging pings

Options for debugging ping abnormalities. Refer to [life of a ping](https://docs.sourcegraph.com/dev/architecture/life-of-a-ping) for the steps in the ping process.

> 1. BigQuery: Query the [update_checks error records](https://console.cloud.google.com/bigquery?sq=839055276916:62219ea9d95d4a49880e661318f419ba) and/or [check the latest pings received](https://console.cloud.google.com/bigquery?sq=839055276916:3c6a5282e66a4f0fac1b958305d7b197) based on installer email admin. 

Reviewer: ```suggestion
1. BigQuery: Query the [update_checks error records](https://console.cloud.google.com/bigquery?sq=839055276916:62219ea9d95d4a49880e661318f419ba) and/or [check the latest pings received](https://console.cloud.google.com/bigquery?sq=839055276916:3c6a5282e66a4f0fac1b958305d7b197). 
```

Depending on the investigation, that piece may not be necessary 🤷

1. Dataflow: Review [Dataflow](https://console.cloud.google.com/dataflow/jobs/us-central1/2020-02-05_10_31_47-13247700157778222556?project=telligentsourcegraph&organizationId=1006954638239): WriteSuccessfulRecords should be full of throughputs and the Failed/Error jobs should be empty of throughputs. 

> 1. Stackdriver (log viewer): [Check the frontend logs](https://console.cloud.google.com/logs/viewer?project=sourcegraph-dev&minLogLevel=0&expandAll=false&customFacets=&limitCustomFacetWidth=true&interval=PT1H&resource=k8s_container%2Fcluster_name%2Fdot-com%2Fnamespace_name%2Fprod%2Fcontainer_name%2Ffrontend), which contain all pings that come through Sourcegraph.com. Use the following the advanced filters to find the pings you're interested in.

Reviewer: ```suggestion
1. Stackdriver (log viewer): [Check Sourcegraph.com's frontend logs](https://console.cloud.google.com/logs/viewer?project=sourcegraph-dev&minLogLevel=0&expandAll=false&customFacets=&limitCustomFacetWidth=true&interval=PT1H&resource=k8s_container%2Fcluster_name%2Fdot-com%2Fnamespace_name%2Fprod%2Fcontainer_name%2Ffrontend) for errors from the [`updatecheck` handler](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/cmd/frontend/internal/app/pkg/updatecheck/handler.go). Use the following advanced filters to find errors you're interested in.
```

> 1. Test on a Sourcegraph dev instance to make sure the pings are being sent properly

Reviewer: Should this line be moved below the filters?
